PolicyResponse de-serialization errors

If you try to de-serialize a Policy response from the Consul ACL Policies API, and the Policy contains data center(s), you get the following exception:

```
com.orbitz.consul.ConsulException: Consul request failed
 at com.orbitz.consul.util.Http.executeCall([Http.java:52](https://kiwiproject-consulclien-o3vp1l54gk2.ws-us97.gitpod.io/))
 at com.orbitz.consul.util.Http.extract([Http.java:31](https://kiwiproject-consulclien-o3vp1l54gk2.ws-us97.gitpod.io/))
 at com.orbitz.consul.AclClient.createPolicy([AclClient.java:75](https://kiwiproject-consulclien-o3vp1l54gk2.ws-us97.gitpod.io/))
 at com.orbitz.consul.AclTest.testCreateAndReadPolicy_WithDatacenters([AclTest.java:77](https://kiwiproject-consulclien-o3vp1l54gk2.ws-us97.gitpod.io/))
Caused by: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize value of type `java.lang.String` from Array value (token `JsonToken.START_ARRAY`)
 at [Source: (okhttp3.ResponseBody$BomAwareReader); line: 6, column: 20] (through reference chain: com.orbitz.consul.model.acl.ImmutablePolicyResponse$Json["Datacenters"])
 at com.fasterxml.jackson.databind.exc.MismatchedInputException.from([MismatchedInputException.java:59](https://kiwiproject-consulclien-o3vp1l54gk2.ws-us97.gitpod.io/))
```
The problem is that the value of `Datacenters` returned by Consul is either `null` or a _list_ of data centers, as strings. But the `BasePolicyResponse` class defines `Datacenters` as:

```java
@JsonProperty("Datacenters")
public abstract Optional<String> datacenters();
```

It _should_ be defined as:

```java
@JsonProperty("Datacenters")
public abstract Optional<List<String>> datacenters();
```

so that it can support the `null` case as well as the case when there is one or more data center. It is unfortunate that Consul returns a `null` value for a collection type, but that's how it is...

--- a/src/itest/java/com/orbitz/consul/AclTest.java
+++ b/src/itest/java/com/orbitz/consul/AclTest.java
@@ -1,18 +1,32 @@
 package com.orbitz.consul;
 
 import com.google.common.net.HostAndPort;
-import com.orbitz.consul.model.acl.*;
+import com.orbitz.consul.model.acl.ImmutablePolicy;
+import com.orbitz.consul.model.acl.ImmutablePolicyLink;
+import com.orbitz.consul.model.acl.ImmutableRole;
+import com.orbitz.consul.model.acl.ImmutableRolePolicyLink;
+import com.orbitz.consul.model.acl.ImmutableToken;
+import com.orbitz.consul.model.acl.PolicyResponse;
+import com.orbitz.consul.model.acl.RoleResponse;
+import com.orbitz.consul.model.acl.Token;
+import com.orbitz.consul.model.acl.TokenResponse;
 
-import org.hamcrest.MatcherAssert;
 import org.junit.BeforeClass;
 import org.junit.Test;
 
 import java.time.Duration;
-import java.util.*;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
 
 import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.core.IsNot.not;
-import static org.junit.Assert.*;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+
 import org.testcontainers.containers.GenericContainer;
 
 public class AclTest {
@@ -62,9 +76,26 @@ public class AclTest {
         String policyName = UUID.randomUUID().toString();
         PolicyResponse policy = aclClient.createPolicy(ImmutablePolicy.builder().name(policyName).build());
         assertThat(policy.name(), is(policyName));
+        assertThat(policy.datacenters(), is(Optional.empty()));
 
         policy = aclClient.readPolicy(policy.id());
         assertThat(policy.name(), is(policyName));
+        assertThat(policy.datacenters(), is(Optional.empty()));
+    }
+
+    @Test
+    public void testCreateAndReadPolicy_WithDatacenters() {
+        AclClient aclClient = client.aclClient();
+
+        String policyName = UUID.randomUUID().toString();
+        ImmutablePolicy newPolicy = ImmutablePolicy.builder().name(policyName).datacenters(List.of("dc1")).build();
+        PolicyResponse policy = aclClient.createPolicy(newPolicy);
+        assertThat(policy.name(), is(policyName));
+        assertThat(policy.datacenters(), is(Optional.of(List.of("dc1"))));
+
+        policy = aclClient.readPolicy(policy.id());
+        assertThat(policy.name(), is(policyName));
+        assertThat(policy.datacenters(), is(Optional.of(List.of("dc1"))));
     }
 
     @Test
@@ -209,7 +240,7 @@ public class AclTest {
                 .description(newDescription)
                 .build();
         TokenResponse updatedToken = aclClient.updateToken(createdToken.accessorId(), tokenUpdates);
-        MatcherAssert.assertThat(updatedToken.description(), is(newDescription));
+        assertThat(updatedToken.description(), is(newDescription));
 
         TokenResponse readToken = aclClient.readToken(createdToken.accessorId());
         assertThat(readToken.description(), is(newDescription));

--- a/src/main/java/com/orbitz/consul/model/acl/BasePolicyResponse.java
+++ b/src/main/java/com/orbitz/consul/model/acl/BasePolicyResponse.java
@@ -3,6 +3,7 @@ package com.orbitz.consul.model.acl;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 import java.math.BigInteger;
+import java.util.List;
 import java.util.Optional;
 
 public abstract class BasePolicyResponse {
@@ -14,7 +15,7 @@ public abstract class BasePolicyResponse {
     public abstract String name();
 
     @JsonProperty("Datacenters")
-    public abstract Optional<String> datacenters();
+    public abstract Optional<List<String>> datacenters();
 
     @JsonProperty("Hash")
     public abstract String hash();

--- a/src/main/java/com/orbitz/consul/model/acl/Policy.java
+++ b/src/main/java/com/orbitz/consul/model/acl/Policy.java
@@ -4,7 +4,6 @@ import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import com.fasterxml.jackson.databind.annotation.JsonSerialize;
-import com.google.common.collect.ImmutableList;
 import org.immutables.value.Value;
 
 import java.util.List;
@@ -29,7 +28,6 @@ public abstract class Policy {
     public abstract Optional<String> rules();
 
     @JsonProperty("Datacenters")
-    @JsonDeserialize(as = ImmutableList.class, contentAs = String.class)
     public abstract Optional<List<String>> datacenters();
 
 }

